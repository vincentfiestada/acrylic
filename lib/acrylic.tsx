/**
 * Re-export all components in one place
 */
export { default as IProps } from "./components/IProps";
export { default as IEventProps } from "./components/IEventProps";

/**
 * Layout Components
 */
export { default as ILayoutProps } from "./components/layout/ILayoutProps";
export { default as Section } from "./components/layout/Section";
export { default as Column } from "./components/layout/Column";
export { default as Row } from "./components/layout/Row";
export { default as Alert } from "./components/layout/Alert";
export { default as Toolbar } from "./components/layout/Toolbar";

/**
 * Typography Components
 */
export { default as ITextProps } from "./components/typography/ITextProps";
export { default as Title } from "./components/typography/Title";
export { default as Text } from "./components/typography/Text";
export { default as Icon } from "./components/typography/Icon";

/**
 * Control Components
 */
export { default as Button } from "./components/controls/Button";
export { default as IconButton } from "./components/controls/IconButton";

/**
 * Variants
 */
export { default as Variant } from "./components/Variant";
