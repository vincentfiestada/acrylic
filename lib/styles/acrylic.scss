@import "normalize.css";
@import "typography";
@import "controls";
