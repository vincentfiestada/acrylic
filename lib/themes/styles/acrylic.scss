@import "normalize.css";
@import "typography";
@import "icons";
@import "layout";
@import "controls";
@import "animation";
