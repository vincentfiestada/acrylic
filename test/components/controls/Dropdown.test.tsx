import * as React from "react";
import { mount } from "enzyme";
import { Dropdown, Button, Menu, MenuItem } from "../../../lib/acrylic";

describe("<Dropdown />", () => {
	it("Should match the latest snapshots", () => {
		expect(
			mount(
				<Dropdown>
					<Button>Dropdown</Button>
					<Menu>
						<MenuItem>Cut</MenuItem>
						<MenuItem>Copy</MenuItem>
						<MenuItem>Paste</MenuItem>
						<MenuItem>Undo</MenuItem>
						<MenuItem disabled>Redo</MenuItem>
					</Menu>
				</Dropdown>
			)
		).toMatchSnapshot();
	});
	it("Should show/hide the menu when clicked", () => {
		const dropdown = mount(
			<Dropdown>
				<Button>Dropdown</Button>
				<Menu />
			</Dropdown>
		);
		expect(
			dropdown
				.find(Menu)
				.parent()
				.hasClass("acr-hidden")
		).toBe(true);
		dropdown.find(Button).simulate("focus");
		expect(
			dropdown
				.find(Menu)
				.parent()
				.hasClass("acr-hidden")
		).toBe(false);
		dropdown.find(Button).simulate("focus");
		expect(
			dropdown
				.find(Menu)
				.parent()
				.hasClass("acr-hidden")
		).toBe(true);
	});
});
