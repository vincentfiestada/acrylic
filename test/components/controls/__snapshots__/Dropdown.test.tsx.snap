// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Should match the latest snapshots 1`] = `
<Dropdown>
  <div
    className="acr-dropdown"
    onClick={[Function]}
  >
    <Button
      onClick={[Function]}
    >
      <button
        className="acr-btn"
        onClick={[Function]}
      >
        Dropdown
      </button>
    </Button>
    <div
      className="acr-hidden"
    >
      <Menu>
        <ul
          className="acr-menu"
        >
          <MenuItem>
            <li
              className="acr-menu-item"
            >
              Cut
            </li>
          </MenuItem>
          <MenuItem>
            <li
              className="acr-menu-item"
            >
              Copy
            </li>
          </MenuItem>
          <MenuItem>
            <li
              className="acr-menu-item"
            >
              Paste
            </li>
          </MenuItem>
          <MenuItem>
            <li
              className="acr-menu-item"
            >
              Undo
            </li>
          </MenuItem>
          <MenuItem
            disabled={true}
          >
            <li
              className="acr-menu-item"
              disabled={true}
            >
              Redo
            </li>
          </MenuItem>
        </ul>
      </Menu>
    </div>
  </div>
</Dropdown>
`;
